### PR TITLE
Repeat Database changes

### PR DESCRIPTION
**Pull request by Shilpi75**
1. repeat daily logic
2. database migration added

### CODE REVIEW

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/AppUtils.java [hunk 1]
```diff
@@ -174,4 +174,12 @@ public static boolean hasUserSeenOnboarding(Context context) {
         return PreferenceManager.getDefaultSharedPreferences(context)
                 .getBoolean(context.getString(R.string.pref_has_user_seen_onboarding), false);
     }
+
+    public static boolean isRepeatableDailyEligible(int repeatCode) {
+        LocalDate today = LocalDate.fromDateFields(new Date());
+        // We want dayOfWeek to start from 0. That is, for Monday: dayOfWeek = 0.
+        int dayOfWeek = today.getDayOfWeek()-1;
+        // TODO: verify this
+        return ((repeatCode & (1 << dayOfWeek)) != 0);
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/AppUtils.java]: * Verified. Remove this `TODO` comment. Also, the variable `dayOfWeek` is a little bit confusing despite the comment. So please change that to `numBitShifts`.
* Move this to `TaskStateUtils`, all the state logic belongs there.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/TaskActionUtils.java [hunk 2]
```diff
@@ -21,7 +22,18 @@
 
     public static void onTaskMarkedDone(@NonNull Context appContext, TaskModel task) {
         task.setLastTriggered(new LocalDate());
-        task.setIsDone(1);
+        if (task.getRepeatType() == DbConstants.REPEAT_DAILY) {
+            LocalDate today = LocalDate.fromDateFields(new Date());
+            if (task.getNextStartDate().compareTo(today) == 0) {
+                task.setIsDone(1);
+            } else {
```
[the remark this answers is not in the document]
Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/TaskActionUtils.java]: correction: not nextStartDate but endDate needs to be compared with Today.

[the remark this answers is not in the document]
Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/TaskActionUtils.java]: Shouldn't it be `endDate.compareTo(today) <= 0` ? Please check.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/database/DbConstants.java [hunk 3]
```diff
@@ -39,6 +40,19 @@
...
+     */
+    public static final int REPEAT_MONDAY = 1;
+    public static final int REPEAT_TUESDAY = 2;
+    public static final int REPEAT_WEDNESDAY = 4;
+    public static final int REPEAT_THURSDAY = 8;
+    public static final int REPEAT_FRIDAY = 16;
+    public static final int REPEAT_SATURDAY = 32;
+    public static final int REPEAT_SUNDAY = 64;
+    public static final int REPEAT_NO_DAY = 0;
+    public static final int REPEAT_ALL_DAYS = 127;
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/database/DbConstants.java]: Won't it help in addition while task creation?

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/TaskModel.java [hunk 4]
```diff
@@ -331,6 +348,8 @@ public void setDateAdded(@NonNull LocalDate dateAdded) {
          */
         private LocalDate nextStartDate = null;
         private int repeatType = DbConstants.NO_REPEAT;
+        // TODO: check this
+        private int repeatCode = DbConstants.REPEAT_NO_DAY;
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/TaskModel.java]: This is necessary. It is not null.

**Pull request merged**